crates/rpc/src/handlers/state.rs

```diff
@@ -1,86 +1,156 @@
+            error::ErrorNotFound(format!("Failed to find `block_root` from {state_id:?}"))
+        })
 }
 
-pub async fn get_state(state_id: ID, db: ReamDB) -> Result<impl Reply, Rejection> {
-    let state = get_state_from_id(state_id, &db).await?;
+#[get("/beacon/states/{state_id}")]
+pub async fn get_beacon_state(
+    db: Data<ReamDB>,
+    state_id: Path<ID>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db).await?;
 
-    Ok(with_status(BeaconResponse::json(state), StatusCode::OK))
+    Ok(HttpResponse::Ok().json(BeaconResponse::new(state)))
 }
 
-pub async fn get_state_root(state_id: ID, db: ReamDB) -> Result<impl Reply, Rejection> {
-    let state = get_state_from_id(state_id, &db).await?;
+#[get("/beacon/states/{state_id}/root")]
+pub async fn get_state_root(
+    db: Data<ReamDB>,
+    state_id: Path<ID>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db).await?;
 
     let state_root = state.tree_hash_root();
 
-    Ok(with_status(
-        BeaconResponse::json(RootResponse::new(state_root)),
-        StatusCode::OK,
-    ))
+    Ok(HttpResponse::Ok().json(BeaconResponse::new(state_root)))
+}
+
+/// Called by `/eth/v1/beacon/states/{state_id}/fork` to get fork of state.
+#[get("/beacon/states/{state_id}/fork")]
+pub async fn get_state_fork(
+    db: Data<ReamDB>,
+    state_id: Path<ID>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db).await?;
+
+    Ok(HttpResponse::Ok().json(BeaconResponse::new(state.fork)))
+}
+
+/// Called by `/states/<state_id>/finality_checkpoints` to get the Checkpoint Data of state.
+#[get("/beacon/states/{state_id}/finality_checkpoints")]
+pub async fn get_state_finality_checkpoint(
+    db: Data<ReamDB>,
+    state_id: Path<ID>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db).await?;
+
+    let response = CheckpointData::new(
+        state.previous_justified_checkpoint,
+        state.current_justified_checkpoint,
+        state.finalized_checkpoint,
+    );
+    Ok(HttpResponse::Ok().json(BeaconResponse::new(response)))
+}
+
+/// Called by `/states/<state_id>/randao` to get the Randao mix of state.
+/// Pass optional `epoch` in the query to get randao for particular epoch,
+/// else will fetch randao of the state epoch
+#[get("/beacon/states/{state_id}/randao")]
+pub async fn get_state_randao(
+    db: Data<ReamDB>,
+    state_id: Path<ID>,
+    query: Json<RandaoQuery>,
+) -> actix_web::Result<impl Responder> {
+    let state = get_state_from_id(state_id.into_inner(), &db).await?;
+
+    let randao_mix = match query.epoch {
+        Some(epoch) => state.get_randao_mix(epoch),
+        None => state.get_randao_mix(state.get_current_epoch()),
+    };
+
+    let response = RandaoResponse { randao: randao_mix };
+    Ok(HttpResponse::Ok().json(BeaconResponse::new(response)))
```

Comment:
```suggestion
    Ok(HttpResponse::Ok().json(BeaconResponse::new(RandaoResponse::new(randao_mix))))
```

Could we inline this?